### PR TITLE
Better error message for invalid bigint literals ending with n

### LINKED ISSUE #28887: Better error message for invalid bigint literals ending with n
## Search Terms

bigint literal error message identifierstart numeric

## Suggestion

@DanielRosenwasser's comment in #28857:

> > `!!! error TS1351: An identifier or keyword cannot immediately follow a numeric literal.`
> This could be better if you want to send a follow-up PR to see if you have a length of 1 and the identifier is named n.
 
## Use Cases

This seems like a good low-hanging fruit for a more specific error message per type of syntax error.

## Examples

The `parseBigInt.ts` baseline test has the following cases:

* `{ const legacyOct = 0123n; }`
* `{ const scientific = 1e2n; }`
* `{ const decimal = 4.1n; }`
* `{ const leadingDecimal = .1n; }`

How about in order...

* `A bigint literal cannot start with a zero.`
* `A bigint literal cannot be a scientific number.`
* `A bigint literal cannot be a decimal number`
* ☝

## Checklist

My suggestion meets these guidelines:

* [x] This wouldn't be a breaking change in existing TypeScript/JavaScript code
* [x] This wouldn't change the runtime behavior of existing JavaScript code
* [x] This could be implemented without emitting different JS based on the types of the expressions
* [x] This isn't a runtime feature (e.g. library functionality, non-ECMAScript syntax with JavaScript output, etc.)
* [x] This feature would agree with the rest of [TypeScript's Design Goals](https://github.com/Microsoft/TypeScript/wiki/TypeScript-Design-Goals).

I can quickly send this in as an addendum to #28857 so they land in the same release.

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
Helps with the cases of decimals and exponentials, though not with numbers incorrectly starting with `0`.

Fixes #28887.

### CODE REVIEW

File: tests/baselines/reference/parseBigInt.errors.txt [hunk 1]
```diff
@@ -82,13 +82,13 @@ tests/cases/compiler/parseBigInt.ts(70,72): error TS2345: Argument of type '3' i
 !!! error TS1005: ',' expected.
     { const scientific = 1e2n; }
```
Comment (JoshuaKGoldberg) [inline tests/baselines/reference/parseBigInt.errors.txt]: You know, on second thought, I think the `~` should cover the number _and_ the identifier. Will switch...

File: src/compiler/scanner.ts [hunk 2]
```diff
@@ -985,20 +985,31 @@ namespace ts {
...
+                if (isScientific) {
+                    error(Diagnostics.A_bigint_literal_cannot_use_exponential_notation, numericStart, identifierStart - numericStart + 1);
+                }
+                else {
+                    error(Diagnostics.A_bigint_literal_must_be_an_integer, numericStart, identifierStart - numericStart + 1);
+                }
+            }
+            else {
+                error(Diagnostics.An_identifier_or_keyword_cannot_immediately_follow_a_numeric_literal, identifierStart, length);
+                pos = identifierStart;
```
Comment (JoshuaKGoldberg) [inline src/compiler/scanner.ts]: Note that `pos` is no longer reset to the beginning of the identifier if it's an `n` part of an invalid bigint literal. Given that the error message indicates it's part of the invalid literal, that seems fine?

File: tests/baselines/reference/identifierStartAfterNumericLiteral.errors.txt [hunk 3]
```diff
@@ -67,15 +69,19 @@ tests/cases/compiler/identifierStartAfterNumericLiteral.ts(26,5): error TS2304:
        ~~
 !!! error TS1351: An identifier or keyword cannot immediately follow a numeric literal.
     3en[null]
```
Comment (DanielRosenwasser) [inline tests/baselines/reference/identifierStartAfterNumericLiteral.errors.txt]: This is a great test case.

**Comment by DanielRosenwasser** [comment]
Thanks!

**Pull request merged**